notebooks/Get Species data from multiple sorces (spocc).qmd

```diff
@@ -0,0 +1,267 @@
+    cat(paste0(pkg, " is already installed and has been loaded!\n"))
+  }
+}
+
+# Install missing packages and load them
+for (pkg in packages) {
+  if (!requireNamespace(pkg, quietly = TRUE)) {
+    install.packages(pkg)
+    cat_message(pkg, "installed")
+  } else {
+    cat_message(pkg, "loading")
+  }
+  library(pkg, character.only = TRUE)
+}
+
+library(spocc)
+library(tidyverse)
+library(leaflet)
+library(sf)
+library(rnaturalearth)
+library(rnaturalearthdata)
+```
+
+## Step 1 Overview and Conceptulisation : Notebook objective Taxon, location, predictors, scale
+
+# Introduction
+
+Shorebirds are key indicators of wetland health. This notebook demonstrates how to use `spocc` to fetch and visualise species occurrence data for:
+
+-   **Bar-tailed Godwit (*Limosa lapponica*)**
+-   **Red-necked Stint (*Calidris ruficollis*)**
+-   **Curlew Sandpiper (*Calidris ferruginea*)**
+
+These are long distance migratory shore birds which make their way to Australia and Tasmania, we will guide you how to check for species data through multiple available data collection.
+
+## Step 2 Get data: Listing our Bird species
+
+We use `spocc` to retrieve up to **500 occurrence records** per species from multiple data sources (GBIF, iNaturalist, OBIS).
+
+```{r}
+# Define species of interest
+species_list <- c("Limosa lapponica", "Calidris ruficollis", "Calidris ferruginea")
+
+# Query multiple sources for species occurrences
+species_data <- lapply(species_list, function(species) {
+  occ(query = species, from = c("gbif", "inat", "obis"), limit = 500)
+})
+
+# Convert results to data frames
+species_dfs <- lapply(species_data, function(data) occ2df(data$gbif))
+
+# Assign species names
+for (i in seq_along(species_dfs)) {
+  species_dfs[[i]]$species <- species_list[i]
+}
+
+# Combine data into one dataframe
+occurrences <- bind_rows(species_dfs)
+
+## Preview Data
+for (i in seq_along(species_list)) {
+  cat("\nPreview of", species_list[i], "data:\n")
+  print(head(species_dfs[[i]], 10))  # Print the first 10 rows of each species' data
+}
+```
+
+## Step 3 Data Cleaning and Filtering
+
+We remove invalid coordinates and filter occurrences within **Australia’s boundary**.
+
+```{r}
+# Remove NA values for latitude and longitude
+occurrences_clean <- occurrences %>%
+  filter(!is.na(longitude) & !is.na(latitude))
+
+# Load world countries and extract Australia
+world <- ne_countries(scale = "medium", returnclass = "sf")
+australia <- world %>% filter(admin == "Australia")
+
+# Convert occurrences to a spatial object
+occurrences_sf <- st_as_sf(occurrences_clean, coords = c("longitude", "latitude"), crs = 4326, remove = FALSE)
+
+# Filter occurrences (approximate using Australia's boundary)
+occurrences_aus <- st_intersection(occurrences_sf, australia)
+
+# Print summary of filtered data
+print(nrow(occurrences_aus))
+print(head(occurrences_aus))
+```
+
+## Step 4 Data Visualisation
+
+We assign **EcoCommons theme colors** for better visualization 😉.
```

Comment:
Recommend replacing this text with something more informative about what you are displaying